dart/lib/src/sentry_envelope_item.dart

```diff
@@ -79,6 +80,20 @@ class SentryEnvelopeItem {
     );
   }
 
+  /// Create an [SentryEnvelopeItem] which holds the [ClientReport] data.
+  factory SentryEnvelopeItem.fromClientReport(ClientReport clientReport) {
+    final bytes = _jsonToBytes(clientReport.toJson());
```

Comment:
I noticed that a few other types use the `_CachedItem` so everything happens kinda in the background, should we do that too? `fromUserFeedback` does not do it, not sure if we have a reason for that.

Reply:
huh, interesting. This was actually changed for user feedback and event in #528 and i used the new approach. Does anyone of you two remember why we moved away from the cached item in those cases? @ueman @marandaneto

Reply:
I don't know anymore, but it probably doesn't make much difference for user feedback as it's not as much information as file attachments, for example.

I think the serialization of the envelopes in general could be significantly improved.

Reply:
`SentryTransaction` and `SentryEvent` are nearly the same object, one uses `_CachedItem` and the other one does not.
I'd rather say, let's use `_CachedItem` for now, and migrate the other ones to a different PR, which does not need to be addressed in this PR.

Reply:
https://github.com/getsentry/sentry-dart/issues/812